src/main/java/io/kestra/plugin/docker/Build.java

```diff
@@ -0,0 +1,218 @@
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.models.tasks.Task;
+import io.kestra.plugin.scripts.exec.scripts.models.DockerOptions;
+import io.kestra.plugin.scripts.exec.scripts.runners.DockerService;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.runners.RunContext;
+
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import javax.validation.constraints.NotNull;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Schema(
+    title = "Build a Docker image and push it to a remote container registry."
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Build and push a Docker image to a registry",
+            code = {"dockerfile: |",
+                "  FROM ubuntu",
+                "  ARG APT_PACKAGES=\"\"",
+                "",
+                "  RUN apt-get update && apt-get install -y --no-install-recommends ${APT_PACKAGES};",
+                "platforms:",
+                "- linux/amd64",
+                "tag: private-registry.io/unit-test:latest",
+                "buildArgs:",
+                "  APT_PACKAGES: curl",
+                "labels:",
+                "  unit-test: \"true\"",
+                "credentials:",
+                "  username: <your-user>",
+                "  password: <your-password>",
+            }
+        ),
+    }
+)
+public class Build extends Task implements RunnableTask<Build.Output> {
+    @Schema(
+        title = "The URI of the Docker API"
+    )
+    @PluginProperty(dynamic = true)
+    private String host;
+
+    @Schema(
+        title = "Credentials to push your image to a container registry."
+    )
+    @PluginProperty
+    private DockerOptions.Credentials credentials;
+
+    @Schema(
+        title = "The contents of your Dockerfile passed as a string, or a path to the Dockerfile"
+    )
+    @PluginProperty(dynamic = true)
+    private String dockerfile;
+
+    @Schema(
+        title = "The target platform for the image e.g. linux/amd64."
+    )
+    @PluginProperty(dynamic = true)
+    private List<String> platforms;
+
+    @Schema(
+        title = "Whether to push the image to a remote container registry."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean push = false;
+
+    @Schema(
+        title = "Always attempt to pull a newer version of the base image."
+    )
+    @PluginProperty(dynamic = true)
+    @Builder.Default
+    private Boolean pull = true;
+
+    @Schema(
+        title = "The tag of this image."
+    )
+    @PluginProperty(dynamic = true)
+    @NotNull
+    private String tag;
```

Comment:
This should be a list to allow setting a specific tag + latest

Reply:
Well, what I understand is that push is different to build.
So maybe we need to separate this in two tasks because double tagging with latest and a version tag is a very common use case and with the current implementation will mandate to build the image twice. And if the second build is in a different worker this can be long.

Note that you can emit one push command per tag, this will be very quick as the second command will send no layer thanks to layer caching.

Reply:
OK found a way to have array, working on it now